proxy/handlers.go

```diff
@@ -185,6 +185,18 @@ func (p *Proxy) shouldSkipAuthentication(r *http.Request) bool {
 		return true
 	}
 
+	if foundPolicy && pol.AllowPublicUnauthenticatedAccess {
+
+		// Sanity Check: Only allow public access if no other whitelists are in place
+		if pol.AllowedDomains == nil && pol.AllowedGroups == nil && pol.AllowedEmails == nil {
+			log.FromRequest(r).Debug().Msg("proxy: skipping authentication for public route")
+			return true
+		}
+
+		log.FromRequest(r).Error().Msg("proxy: route marked as public contains whitelists, " +
+			"disabling public access and using whitelists")
```

Comment:
Alternatively, this check can be placed in [internal\policy\policy.go::validate](https://github.com/pomerium/pomerium/blob/409f44a9be1d8bbdf768df2b508b7d427d279f72/internal/policy/policy.go#L37). Would it make more sense to do this validation when the server is started and just completely disallow both options from being enabled? I admit it does seem weird running this check in the dataplane when we could have ran it at startup in the controlplane

Reply:
Yeah, I think it would more sense we should pop an error when loading the policy instead. Saves resources, and avoids any user confusion.

Plus, I was thinking that at some point it would be nice to be able to abstract out a the policy validate logic as a stand-alone linter so admins can check their policy prior to any updates.